src/services/FrogbotService.ts

```diff
@@ -0,0 +1,166 @@
+            ]);
+
+            await this.addFrogbotWorkflows(owner, repo.name, defaultBranch);
+            await this.openPullRequest(owner, repo.name, defaultBranch);
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    private async setUpFrogbotEnvironment(owner: string, repo: string) {
+        //TODO: add reviewers funcionality
+        try {
+            await this.octokit.rest.repos.createOrUpdateEnvironment({
+                owner,
+                repo,
+                environment_name: "frogbot"
+            });
+        } catch (error) {
+            console.error('Error adding maintainers to environment:', error);
+        }
+    }
+
+    private async allowWorkflows(owner: string, repo: string) {
+
+        try {
+            await this.octokit.request(`PUT /repos/${owner}/${repo}/actions/permissions`, {
+                owner,
+                repo,
+                enabled: true,
+            });
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    private async createBranch(owner: string, repo: string, defaultBranch: string) {
+        try {
+            const { data: refData } = await this.octokit.rest.git.getRef({
+                owner,
+                repo,
+                ref: `heads/${defaultBranch}`,
+            });
+
+            const latestCommitSha = refData.object.sha;
+            const newBranchRef = `refs/heads/${PULL_REQUEST_DATA.branchName}`;
+            await this.octokit.rest.git.createRef({
+                owner,
+                repo,
+                ref: newBranchRef,
+                sha: latestCommitSha,
+            });
+        } catch (error) {
+            console.error('Error creating branch:', error);
+        }
+    }
+
+    private async addFrogbotWorkflows(owner: string, repo: string, defaultBranch: string) {
+        try {
+            const scanRepoWorkflow = this.scanRepositoryWorkflow(defaultBranch);
+            const pullRequestWorkflow = this.pullRequestWorkflow();
+
+            await Promise.all([
+                this.octokit.rest.repos.createOrUpdateFileContents({
+                    owner,
+                    repo,
+                    path: '.github/workflows/frogbot-scan-repository.yml',
+                    message: `Added frogbot-scan-repository.yml on ${PULL_REQUEST_DATA.branchName}`,
+                    content: Buffer.from(scanRepoWorkflow).toString('base64'),
+                    branch: PULL_REQUEST_DATA.branchName,
+                }),
+                this.octokit.rest.repos.createOrUpdateFileContents({
+                    owner,
+                    repo,
+                    path: '.github/workflows/frogbot-scan-pull-request.yml',
+                    message: `Added frogbot-scan-pull-request.yml on ${PULL_REQUEST_DATA.branchName}`,
+                    content: Buffer.from(pullRequestWorkflow).toString('base64'),
+                    branch: PULL_REQUEST_DATA.branchName,
+                }),
+            ]);
```

Comment:
```suggestion
            await Promise.all([
                this.octokit.rest.repos.createOrUpdateFileContents({
                    owner,
                    repo,
                    path: '.github/workflows/frogbot-scan-repository.yml',
                    message: `Added frogbot-scan-repository.yml on ${PULL_REQUEST_DATA.branchName}`,
                    content: Buffer.from(this.scanRepositoryWorkflow(defaultBranch)).toString('base64'),
                    branch: PULL_REQUEST_DATA.branchName,
                }),
                this.octokit.rest.repos.createOrUpdateFileContents({
                    owner,
                    repo,
                    path: '.github/workflows/frogbot-scan-pull-request.yml',
                    message: `Added frogbot-scan-pull-request.yml on ${PULL_REQUEST_DATA.branchName}`,
                    content: Buffer.from(this.pullRequestWorkflow()).toString('base64'),
                    branch: PULL_REQUEST_DATA.branchName,
                }),
            ]);
```
vars with values from methods. only used once